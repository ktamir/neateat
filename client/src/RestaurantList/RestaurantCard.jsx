// @flow

import { Card } from 'antd';
import React from 'react';
import type { Restaurant } from '../flowTypes';

import './RestaurantCard.scss';

type Props = {|
  restaurant: Restaurant
|}

export const RestaurantCard = (props: Props) => {
  const { name, accepts10Bis, rating } = props.restaurant;
  return (
    <Card className="restaurant-card">
      <div className="card-details">
        <h3>{name}</h3>
        {accepts10Bis && <span className="ten-bis">10bis</span>}
        <div>Rating: {Array.from({ length: rating }, (_, i) =>
          <span aria-label="star" role="img" key={i}>‚≠ê</span>)}</div>
      </div>
    </Card>
  );
};
