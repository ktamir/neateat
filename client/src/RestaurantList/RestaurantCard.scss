.restaurant-card {
  width: 300px;
  height: 100px;
  h3 {
    display: inline;
  }
  margin: 15px;
  .ten-bis {
    color: gray;
    font-size: 9px;
    display: inline;
    margin-left: 5px;
  }
  .card-details {
    margin-left: 20px;
  }
}
