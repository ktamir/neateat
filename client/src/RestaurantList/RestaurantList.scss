 .restaurant-fetch-error {
    margin-top: 20px;
  }
