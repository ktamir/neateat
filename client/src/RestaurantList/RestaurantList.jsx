// @flow
import React, { Component } from 'react';
import { RestaurantCard } from './RestaurantCard';
import type { Restaurant } from '../flowTypes';

type Props = { restaurants: Array<Restaurant> }

class RestaurantList extends Component<Props> {
  render() {
    return (
      <div className="restaurant-list">
        {this.props.restaurants.map(restaurant => <RestaurantCard key={restaurant.id} restaurant={restaurant}/>)}
      </div>
    );
  }
}

export default RestaurantList;
