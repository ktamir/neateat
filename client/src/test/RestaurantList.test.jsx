import { mount, shallow } from 'enzyme/build';
import { RestaurantList } from '../RestaurantList/RestaurantList';
import React from 'react';
import { mockRestaurants } from './testConsts';

describe('RestaurantList', () => {
  const fetchRestaurants = jest.fn();

  beforeEach(() => {
    fetchRestaurants.mockClear();
  });

  it('should trigger fetch request', () => {
    mount(<RestaurantList restaurants={[]} fetchRestaurants={fetchRestaurants}
      isLoading={false} />);
    expect(fetchRestaurants).toHaveBeenCalledTimes(1);
  });

  it('should render restaurants correctly', () => {
    let restaurantCard = shallow(<RestaurantList restaurants={mockRestaurants} fetchRestaurants={fetchRestaurants}
      isLoading={false} />);
    expect(restaurantCard).toMatchSnapshot();
  });

  it('should render loading correctly', () => {
    let restaurantCard = shallow(<RestaurantList restaurants={[]} fetchRestaurants={fetchRestaurants}
      isLoading={true} />);
    expect(restaurantCard).toMatchSnapshot();
  });
});
