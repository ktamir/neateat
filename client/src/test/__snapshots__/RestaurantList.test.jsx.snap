// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantList should render loading correctly 1`] = `
<div
  className="restaurant-list"
>
  <Spin
    size="default"
    spinning={true}
    tip="Loading..."
    wrapperClassName=""
  />
</div>
`;

exports[`RestaurantList should render restaurants correctly 1`] = `
<div
  className="restaurant-list"
>
  <RestaurantCard
    key="1"
    restaurant={
      Object {
        "accepts10Bis": true,
        "address": "144 Derech Menachem Begin, Tel Aviv, Israel",
        "cuisine": "French",
        "id": 1,
        "maxDeliveryTime": 60,
        "name": "Chez Tamir",
        "rating": 2,
      }
    }
  />
  <RestaurantCard
    key="2"
    restaurant={
      Object {
        "accepts10Bis": true,
        "address": "7 Derech Namir, Tel Aviv, Israel",
        "cuisine": "Italian",
        "id": 2,
        "maxDeliveryTime": 100,
        "name": "Tamir Trattoria",
        "rating": 1,
      }
    }
  />
</div>
`;
